### PR TITLE
Updated data pipeline documentation with troubleshooting instructions.

### PR DESCRIPTION
**Pull request by chintalavr**
Updated data pipeline documentation with troubleshooting instructions to address Orders database permissions when using MSAs to access dashboard reports.

### CODE REVIEW

File: docs/azure_jumpstart_ag/contoso_supermarket/data_pos/_index.md [hunk 1]
```diff
@@ -117,3 +117,28 @@ By default there is no data available in Cosmos DB database after the deployment
 ## Next steps
 
 Now that you have completed the first data pipeline scenario, it's time to continue to the next scenario, [Data pipeline and reporting across cloud and edge for sensor telemetry](https://azurearcjumpstart.io/azure_jumpstart_ag/contoso_supermarket/freezer_monitor/).
+
+## Troubleshooting
+
+### Principal 'msauser=xyz@abc.com' is not authorized to read database Orders
+
+Depending on the type of user account being used to access ADX dashboards, you might have issues accessing data in Orders database in ADX cluster through ADX dashboard reports. Microsoft Accounts (MSAs) are all of the Microsoft-managed non-organizational user accounts. For example, hotmail.com, live.com, outlook.com. These MSAs require special syntax to grant database access permissions in ADX cluster. Refer [Referencing security principals](https://learn.microsoft.com/en-us/azure/data-explorer/kusto/management/referencing-security-principals#microsoft-accounts-msas) to use correct syntax to grant user permissions to the ADX database.
```
Comment (likamrat) [inline docs/azure_jumpstart_ag/contoso_supermarket/data_pos/_index.md:125]:
```suggestion
Depending on the type of user account being used to access ADX dashboards, you might have issues accessing data in the _Orders_ database in the ADX cluster through ADX dashboard reports. Microsoft Accounts (MSAs) are all of the Microsoft-managed non-organizational user accounts. For example, **_hotmail.com, live.com, outlook.com_**. These MSAs require special syntax to grant database access permissions in the ADX cluster. Refer to [Referencing security principals](https://learn.microsoft.com/azure/data-explorer/kusto/management/referencing-security-principals#microsoft-accounts-msas) to use the correct syntax to grant user permissions to the ADX database.
```

File: docs/azure_jumpstart_ag/contoso_supermarket/data_pos/_index.md [hunk 2]
```diff
@@ -117,3 +117,28 @@ By default there is no data available in Cosmos DB database after the deployment
...
+
+Screenshot below shows permissions error when using MSAs.
```
Comment (likamrat) [inline docs/azure_jumpstart_ag/contoso_supermarket/data_pos/_index.md:127]:
```suggestion
The screenshot below shows a permissions error when using MSAs.
```

File: docs/azure_jumpstart_ag/contoso_supermarket/data_pos/_index.md [hunk 3]
```diff
@@ -117,3 +117,28 @@ By default there is no data available in Cosmos DB database after the deployment
...
+
+  ![Screenshot showing the principal not authorized to read database error](./img/adx-principal-not-authorized.png)
+
+Follow the steps below to address this permissions error.
```
Comment (likamrat) [inline docs/azure_jumpstart_ag/contoso_supermarket/data_pos/_index.md:131]:
```suggestion
Follow the below steps to address this permissions error.
```

File: docs/azure_jumpstart_ag/contoso_supermarket/data_pos/_index.md [hunk 4]
```diff
@@ -117,3 +117,28 @@ By default there is no data available in Cosmos DB database after the deployment
...
+
+- In the [Azure Portal](https://portal.azure.com/), locate ADX cluster deployed in the resource group and open.
```
Comment (likamrat) [inline docs/azure_jumpstart_ag/contoso_supermarket/data_pos/_index.md:133]:
```suggestion
- In the [Azure portal](https://portal.azure.com/), locate the ADX cluster deployed in the resource group and open.
```

File: docs/azure_jumpstart_ag/contoso_supermarket/data_pos/_index.md [hunk 5]
```diff
@@ -117,3 +117,28 @@ By default there is no data available in Cosmos DB database after the deployment
...
+- Click on Query under Data, select Orders database and enter Kusto query as shown below to grant user access to Orders database. Replace user principal with correct principal to grant permissions.
```
Comment (likamrat) [inline docs/azure_jumpstart_ag/contoso_supermarket/data_pos/_index.md:134]:
```suggestion
- Click on _Query_ under Data, select the _Orders_ database, and enter the Kusto query as shown below to grant user access to the _Orders_ database. Replace the user principal with the correct principal to grant permissions.
```

File: docs/azure_jumpstart_ag/contoso_supermarket/data_pos/_index.md [hunk 6]
```diff
@@ -117,3 +117,28 @@ By default there is no data available in Cosmos DB database after the deployment
...
+
+  ```
+  .add database Orders users ('msauser=xyz@hotmail.com') 'XYZ (hotmail.com)'
```
Comment (likamrat) [inline docs/azure_jumpstart_ag/contoso_supermarket/data_pos/_index.md:137]: not sure but is there a specific kql code block syntax needed? 

For example

```shel
code
```

File: docs/azure_jumpstart_ag/contoso_supermarket/data_pos/_index.md [hunk 7]
```diff
@@ -117,3 +117,28 @@ By default there is no data available in Cosmos DB database after the deployment
...
+  ```
+
+- Click Run to execute Kusto query to grant permissions
```
Comment (likamrat) [inline docs/azure_jumpstart_ag/contoso_supermarket/data_pos/_index.md:140]:
```suggestion
- Click _Run_ to execute the Kusto query to grant permissions.
```

File: docs/azure_jumpstart_ag/contoso_supermarket/data_pos/_index.md [hunk 8]
```diff
@@ -117,3 +117,28 @@ By default there is no data available in Cosmos DB database after the deployment
...
+
+  ![Screenshot showing how to grant user permissions](./img/adx-database-grant-user-access.png)
+
+- Once user permission is granted go to [ADX dashboards](https://dataexplorer.azure.com/dashboards) and refresh dashboard report to view Orders data.
```
Comment (likamrat) [inline docs/azure_jumpstart_ag/contoso_supermarket/data_pos/_index.md:144]:
```suggestion
- Once user permission is granted go to [ADX dashboards](https://dataexplorer.azure.com/dashboards) and refresh the dashboard report to view _Orders_ data.
```

**Review by likamrat — CHANGES_REQUESTED** [review]
- This should also be included in the sensors scenario
- Multiple sytax comments

File: docs/azure_jumpstart_ag/contoso_supermarket/data_pos/_index.md (hunk 6 above)
Comment (chintalavr) [reply docs/azure_jumpstart_ag/contoso_supermarket/data_pos/_index.md:137]: I usually consider SQL or KQL as a code. Anything that can differentiate between documentation vs syntax will be helpful.

File: docs/azure_jumpstart_ag/contoso_supermarket/freezer_monitor/_index.md [hunk 9]
```diff
@@ -76,6 +76,8 @@ To view the Freezer Monitoring dashboard you will first need to import it into A
 
   ![Screenshot showing the default freezer dashboard](./img/adx_freezer_dashboard_default.png)
 
+  > __NOTE: Depending on the type of user account being used to access ADX dashboards, you might have issues accessing data in the _Orders_ database in the ADX cluster with an error _User principal 'msauser=xyz@abc.com' is not authorized to read database 'Orders'_. If you experience this access issue, refer [Jumpstart Agora - Contoso Supermarket scenario troubleshooting](https://azurearcjumpstart.io/azure_jumpstart_ag/contoso_supermarket/troubleshooting#user-principal-is-not-authorized-to-read-database-orders) guide to troubleshoot and address this access issue__.
```
Comment (likamrat) [inline docs/azure_jumpstart_ag/contoso_supermarket/freezer_monitor/_index.md:79]:
```suggestion
  > __NOTE: Depending on the type of user account being used to access ADX dashboards, you might have issues accessing data in the _Orders_ database in the ADX cluster with an error _User principal 'msauser=xyz@abc.com' is not authorized to read database 'Orders'_. If you experience this access issue, refer to [Jumpstart Agora - Contoso Supermarket scenario troubleshooting](https://azurearcjumpstart.io/azure_jumpstart_ag/contoso_supermarket/troubleshooting#user-principal-is-not-authorized-to-read-database-orders) guide to troubleshoot and address this access issue__.
```

**Review by likamrat — APPROVED** [review]
LGTM

**Pull request merged**